### ISSUE TITLE
COLMAP with CUDA-enabled BA

### ISSUE THREAD

**Issue by alancneves** [colmap/colmap#2746]
**Describe the bug**
When COLMAP tries to do the BA using CUDA (with _--Mapper.ba_use_gpu option_) it raises a ceres error on `detect_structure.cc` at line 111, indicating empty rows on the matrix.

**To Reproduce**
1. Execute the usual colmap pipeline: extract the features, do the matching.
2. At mapper step, use the option flag `--Mapper.ba_use_gpu on`
> colmap mapper --Mapper.ba_use_gpu on --database_path ~/data/sfm/street_pole1/sparse/database.db --image_path ~/data/sfm/street_pole1/images --output_path ~/data/sfm/street_pole1/sparse
3. When the number of registered files reach 50 files, the retriangulation and Global BA happens.
4. The error triggers with message:
> F0903 18:59:56.927320 76973 detect_structure.cc:111] Check failed: *row_block_size != 0 (0 vs. 0) No rows found

**Expected behavior**
It was expected to run without erros, given that the same dataset without GPU flags runs smooth.

**Screenshots**
![error](https://github.com/user-attachments/assets/7d78fbf4-e180-4f4a-8864-c5ea987b671a)
![error_gdb](https://github.com/user-attachments/assets/97f0ef11-b212-4c0e-a8e2-876332929aa1)

**Environment:**
 - OS: Ubuntu 22.04
 - COLMAP Version: 4ccfeb55e6548a2d15636796e42e1df292c073c5
 - Capture Device: Samsung S23
 - Ceres version: 2.2.0

**Related issue**: https://github.com/colmap/colmap/issues/2643
**Related PR:** https://github.com/colmap/colmap/pull/2717

**Comment by xiemeilong** [comment]
In my case, the process aborted in DetectStructure, and then it was blocked. Here's the log:
```
I20240904 10:01:53.543553 131060093902848 incremental_mapper.cc:379] Registering image #3194 (50)
I20240904 10:01:53.543570 131060093902848 incremental_mapper.cc:382] => Image sees 4066 / 6019 points
I20240904 10:01:54.244916 131060093902848 incremental_mapper.cc:379] Registering image #3195 (51)
I20240904 10:01:54.244934 131060093902848 incremental_mapper.cc:382] => Image sees 3548 / 5662 points
I20240904 10:01:54.785549 131060093902848 incremental_mapper.cc:41] Retriangulation and Global bundle adjustment
*** Aborted at 1725415315 (unix time) try "date -d @1725415315" if you are using GNU date ***
PC: @     0x5776ce43029c ceres::internal::DetectStructure(ceres::internal::CompressedRowBlockStructure const&, int, int*, int*, int*)
*** SIGBUS (@0x0) received by PID 1224412 (TID 0x7732ce576000) from PID 0; stack trace: ***
    @     0x7732fa8d1d99 void std::__invoke_impl<void, void (*)(int, siginfo_t*, void*), int&, siginfo_t*&, void*&>(std::__invoke_other, void (*&&)(int, siginfo_t*, void*), int&, siginfo_t*&, void*&)
    @     0x7732fa8d1bd0 std::__invoke_result<void (*)(int, siginfo_t*, void*), int&, siginfo_t*&, void*&>::type std::__invoke<void (*)(int, siginfo_t*, void*), int&, siginfo_t*&, void*&>(void (*&&)(int, siginfo_t*, void*), int&, siginfo_t*&, void*&)
    @     0x7732fa8d19a6 std::call_once<void (*)(int, siginfo_t*, void*), int&, siginfo_t*&, void*&>(std::once_flag&, void (*&&)(int, siginfo_t*, void*), int&, siginfo_t*&, void*&)::{lambda()#1}::operator()() const

```

Environment:

OS: Ubuntu 24.04
COLMAP Version: https://github.com/colmap/colmap/tree/4ccfeb55e6548a2d15636796e42e1df292c073c5
Capture Device: GoPro
Ceres version: 2.2.0
eigen：3.4.0
SuiteSparse：7.8.2

**Comment by whuaegeanse** [comment]
This should be a bug triggered by the ceres library

**Comment by whuaegeanse** [comment]
Fixed in #2758
